### PR TITLE
Update failing tests after new transient in WP core

### LINKED ISSUE #95: Update trunk tests to accommodate for new transient in core
## Bug Report

- [x] Yes, I reviewed the [contribution guidelines](https://make.wordpress.org/cli/handbook/contributing/).
- [x] Yes, more specifically, I reviewed the guidelines on [how to write clear bug reports](https://make.wordpress.org/cli/handbook/bug-reports/).

**Describe the current, buggy behavior**

https://core.trac.wordpress.org/changeset/58025 was merged last week, which added a new transient.

That now causes some tests to fail, see https://github.com/wp-cli/cache-command/actions/runs/8845496490/job/24289764622?pr=89#step:11:88

**Describe how other contributors can replicate this bug**

Run Behat tests with `WP_VERSION=trunk`

**Describe what you would expect as the correct outcome**

Passing tests :)

**Let us know what environment you are running this on**

Any environment

**Provide a possible solution**

We could force-delete `wp_theme_files_patterns` in the tests, or add some new tests specifically targeting 6.6+, and only run the existing ones on <= 6.5. We could also modify the tests a little bit to be less strict (e.g. use "contain" vs "be")

**Provide additional context/Screenshots**

See link above

### PR DESCRIPTION
**Pull request by petitphp**
This PR update transient feature tests to handle the new theme pattern transient `wp_theme_files_patterns`.

Fixes #95

### CODE REVIEW

**Comment by swissspidy** [comment]
Looks like there are still some failing tests.

What's with the `WP_DEVELOPMENT_MODE` addition? From what I can see [in the commit](https://core.trac.wordpress.org/changeset/58025), the transient is added regardless of the development mode setting.

**Comment by petitphp** [comment]
The method `set_pattern_cache` which internally set the transient is only called if not in development mode : [wp-includes/class-wp-theme.php#L1853-L1855](https://github.com/WordPress/wordpress-develop/blob/81882ddcc043e1ff82e44e89e33878dcd4470a18/src/wp-includes/class-wp-theme.php#L1853-L1855)
```php
if ( $can_use_cached ) {
	$this->set_pattern_cache( $pattern_data );
}
```

**Comment by swissspidy** [comment]
Got it! In that case let's add a comment on these lines explaining what it does and why it is added.

An alternative would be to create a new test for 6.6+, but that is probably overkill.

**Comment by petitphp** [comment]
Will update the PR with comments. I'm also looking at tests that fail with MySQL 5.x

File: features/transient.feature [hunk 1]
```diff
@@ -465,7 +467,7 @@ Feature: Manage WordPress transient cache
       """
 
     When I run `wp transient list --network --format=csv`
-    Then STDOUT should be:
+    Then STDOUT should contain:
```
Comment (swissspidy) [inline features/transient.feature:482]: Maybe doing something like this helps with the test failures:

```
    When I run `wp transient list --network --format=csv`
    Then STDOUT should contain:
      """
      foo4,bar4,false
      """
    And STDOUT should contain:
      """
      foo5,bar5,95649119999
      """
    And STDOUT should contain:
      """
      foo6,bar6,1321009871
      """
```

Comment (petitphp) [reply features/transient.feature:482]: That's a more robust way to fix the issue. I've updated the tests.

**Review by swissspidy — APPROVED** [review]
Seems to work now!

Current SQLite failures are known, see #92

**Comment by swissspidy** [comment]
Just noting this was merged during the [hack day](https://github.com/wp-cli/wp-cli/issues/5935)!

**Pull request merged**